File: spare_parts_shop_flutter/lib/models/payment.dart
import 'customer.dart';
import 'bill.dart';

class Payment {
  final int id;
  final Customer customer;
  final Bill? bill;
  final double amount;
  final String paymentDate;
  final String? note;

  Payment({
    required this.id,
    required this.customer,
    this.bill,
    required this.amount,
    required this.paymentDate,
    this.note,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      id: json['id'],
      customer: Customer.fromJson(json['customer']),
      bill: json['bill'] != null ? Bill.fromJson(json['bill']) : null,
      amount: (json['amount'] as num).toDouble(),
      paymentDate: json['paymentDate'],
      note: json['note'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'customer': customer.toJson(),
      'bill': bill?.toJson(),
      'amount': amount,
      'paymentDate': paymentDate,
      'note': note,
    };
  }
}

File: spare-parts-shop/src/main/java/com/spareparts/model/Payment.java
package com.spareparts.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "payments")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "business_id", nullable = false)
    @JsonIgnore
    private Business business;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "customer_id", nullable = false)
    private Customer customer;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "bill_id")
    @JsonIgnoreProperties({"items", "customer"})
    private Bill bill;

    @Column(nullable = false)
    private Double amount;

    @Column(name = "payment_date", nullable = false)
    private LocalDateTime paymentDate = LocalDateTime.now();

    @Column(columnDefinition = "TEXT")
    private String note;
}
